src/twigextensions/FetchTwigExtension.php

```diff
@@ -1,65 +1,74 @@
 <?php
 /**
- * Fetch plugin for Craft CMS 3.x
+ * Fetch plugin for Craft 4
  *
- * Utilise the Guzzle HTTP client from within your Craft templates.
+ * Guzzle HTTP client from within your Craft templates.
  *
  * @link      https://github.com/jalendport
  * @copyright Copyright (c) 2018 Jalen Davenport
  */
 
 namespace jalendport\fetch\twigextensions;
 
-use jalendport\fetch\Fetch;
-
 use Craft;
+use GuzzleHttp\Exception\GuzzleException;
+use Twig\Extension\AbstractExtension;
+use Twig\TwigFunction;
+use craft\helpers\Json;
 
 /**
-* @author    Jalen Davenport
-* @package   Fetch
-* @since     1.1.0
-*/
-class FetchTwigExtension extends \Twig_Extension
+ * @author    Jalen Davenport
+ * @package   Fetch
+ * @since     1.1.0
+ */
+class FetchTwigExtension extends AbstractExtension
 {
-  public function getName()
-  {
-      return 'Fetch';
-  }
-
-  public function getFunctions()
-  {
-      return [
-          new \Twig_SimpleFunction('fetch', [$this, 'fetch']),
-      ];
-  }
+    /**
+     * @return TwigFunction[]
+     */
+    public function getFunctions(): array
+    {
+        return [
+            new TwigFunction('fetch', [$this, 'fetch']),
+        ];
+    }
 
-  public function fetch($client, $method, $destination, $request = [], $parseJson = true)
-  {
-      $client = new \GuzzleHttp\Client($client);
+    /**
+     * @param $client
+     * @param $method
+     * @param $destination
+     * @param array $request
+     * @param bool $parseJson
+     * @return array
+     */
+    public function fetch($client, $method, $destination, array $request = [], bool $parseJson = true): array
+    {
+        $client = Craft::createGuzzleClient($client);
 
-      try {
+        try {
+            $result = $client->request($method, $destination, $request);
```

Comment:
or if you prefer running it async...
$result = $client->requestAsync($method, $destination, $request)->wait();